src/execution_plan/execution_plan.h

```diff
@@ -65,6 +69,10 @@ OpBase *ExecutionPlan_LocateLastOp(OpBase *root, OPType type);
  * Returns an array of operations. */
 OpBase **ExecutionPlan_LocateOps(OpBase *root, OPType type);
 
+/* Locate all operations matching the given types in the op tree.
+ * Returns an array of operations. */
+OpBase **ExecutionPlan_LocateOpsMatchingType(OpBase *root, const OPType *types, uint type_count);
+
```

Comment:
Moved code as recommended, but the call refactor you suggest would result in incorrect behavior.

`ExecutionPlan_LocateOpMatchingType` is actually a parallel to `ExecutionPlan_LocateFirstOp` - I don't want to use `ExecutionPlan_LocateFirstOpMatchingType` because it's clumsy, and `ExecutionPlan_LocateLastOp` is currently unused.

The `LocateOps` family collects _all_ matching ops in a tree, while the `LocateOp` family returns the first match in a DFS.

Yes, these naming conventions are unintuitive and getting out of hand!

Reply:
Alright then please remove unused function(s) see if you can come up with function names which are a bit clearer.